hazelcast/src/main/java/com/hazelcast/spi/impl/operationexecutor/impl/AltoOperationQueue.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.spi.impl.operationexecutor.impl;
+
+import com.hazelcast.internal.tpc.Reactor;
+import org.jctools.queues.MpmcArrayQueue;
+
+public class AltoOperationQueue implements OperationQueue {
+
+    public static final int CAPACITY = 1024;
+    private Reactor reactor;
+    private MpmcArrayQueue normalQueue = new MpmcArrayQueue(CAPACITY);
+    private MpmcArrayQueue priorityQueue = new MpmcArrayQueue(CAPACITY);
+
+    public AltoOperationQueue(Reactor reactor) {
+        this.reactor = reactor;
+    }
+
+    @Override
+    public void add(Object task, boolean priority) {
+        if (priority) {
+            priorityQueue.offer(task);
```

Comment:
Good one. One of the things I need to think about (back pressure).

Reply:
I have provided exactly the same problematic behavior as HZ classic. We can apply back pressure to clients, but not to members since we can't just block the socket from reading.